core/dbt/task/init.py

```diff
@@ -221,6 +221,13 @@ def create_profile_using_project_profile_template(self, profile_name):
     def ask_for_adapter_choice(self) -> str:
         """Ask the user which adapter (database) they'd like to use."""
         available_adapters = list(_get_adapter_plugin_names())
+
+        if not len(available_adapters):
+            click.echo(
+                "No adapters available. Learn how to install an adapter by going to https://docs.getdbt.com/docs/supported-data-platforms#adapter-installation"
+            )
+            sys.exit(1)
```

Comment:
Resolved in https://github.com/dbt-labs/dbt-core/pull/7609/commits/83cda4ca0498b2a40908fd1d082d6bc6f4b4b05c via a custom dbt exception rather than a `ClickException`

Can revisit as-needed.